evm-contracts/src/v0.7/dev/Operator.sol

```diff
@@ -49,7 +51,8 @@ contract Operator is
   );
 
   event OracleResponse(
-    bytes32 indexed requestId
+    bytes32 indexed requestId,
+    uint256 dataVersion
```

Comment:
@se3000 we do emit the `dataVersion` in the `OracleRequest` event right now. I agree, however, that I'm not sure what it gains any party other than a visible audit trail.

With regards to the `dataVersion` check, happy to add this in, but it will introduce changes to how the `ChainlinkClient` contract manages it. At the moment it's [stored as a constant in `ChainlinkClient`](https://github.com/smartcontractkit/chainlink/blob/develop/evm-contracts/src/v0.6/ChainlinkClient.sol#L23), and passed as a parameter upon making a request.

What's the best approach here? My hunch is to create a new v0.7 `ChainlinkClient`, and simply increase the `ARGS_VERSION` constant. Is that enough or do we want something more complex?

Reply:
Ah yeah, good call. My understanding of the code evolved, and I updated some of the comment but not all. Yes, we emit the data version in the request, but I still don't think it gains us anything in the response event.

You're right that we will need to change the client to have a request that uses data version 2. That can be a separate method with a separate constant. Assuming the inheriting contract only uses one of the two request methods, the optimizer will get rid of the other, so having two in the library shouldn't be a problem.

I think the dataVersion validation should remain in the oracle contract though. Atm, the client reverting can't prevent the oracle from getting paid. This is a guarantee the oracle needs to make to the client, in practice it won't be hard to uphold.